### PR TITLE
Wrapper utility

### PR DESCRIPTION
**Pull request by ExE-Boss**
Split off from #235.

### Adds the following:
- [x] Mod loading progress bar when launching Minecraft
- [x] `DirectionConverter` for converting between NOVA’s `Direction` and Minecraft’s `EnumFacing`
- [x] `WrapUtility.getNovaPlayer` and `WrapUtility.getMCPlayer`

### CODE REVIEW

File: minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/launcher/NovaMinecraft.java [hunk 1]
```diff
@@ -142,7 +143,11 @@ public void preInit(FMLPreInitializationEvent evt) {
 				e.printStackTrace();
 			}
 
-			launcher.load();
+			ProgressManager.ProgressBar progressBar = ProgressManager.push("Loading NOVA mods", modClasses.isEmpty() ? 1 : modClasses.size());
+			launcher.load(new FMLProgressBar(progressBar));
+			while (progressBar.getStep() < progressBar.getSteps())
+				progressBar.step("");
```
Comment (RX14) [inline minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/launcher/NovaMinecraft.java]: Please refactor this into a `finish()` method on `ProgressBar`.

File: minecraft/1.8/src/test/java/nova/core/wrapper/mc/forge/v18/wrapper/VectorConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	VectorConverter converter;
+
+	@Before
+	public void setUp() {
+		converter = new VectorConverter();
+	}
+
+	@Test
+	public void testToNova() {
+		assertThat(converter.toNova(BlockPos.ORIGIN)).isEqualTo(Vector3D.ZERO);
```
Comment (RX14) [inline minecraft/1.8/src/test/java/nova/core/wrapper/mc/forge/v18/wrapper/VectorConverterTest.java]: This needs more test cases, just one or two for negative values, and making sure values don't get swapped around.

File: minecraft/1.8/src/test/java/nova/core/wrapper/mc/forge/v18/wrapper/VectorConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	@Test
+	public void testToNative() {
+		assertThat(converter.toNative(Vector3D.ZERO)).isEqualTo(BlockPos.ORIGIN);
```
Comment (RX14) [inline minecraft/1.8/src/test/java/nova/core/wrapper/mc/forge/v18/wrapper/VectorConverterTest.java]: ditto

File: src/test/java/nova/core/util/ProgressBarTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	EventBus<Event> events;
+
+    @Before
+    public void setUp() {
+		this.events = new EventBus<>();
+	}
+
+	@Test
+	public void testProgressBar() {
+		events.on(ProgressBarEvent.class).bind(evt -> System.out.println(evt.message));
```
Comment (RX14) [inline src/test/java/nova/core/util/ProgressBarTest.java]: This test should add `evt.message` to a List and verify the results with assertions.

File: src/test/java/nova/core/util/ProgressBarTest.java [hunk 5]
```diff
@@ -40,30 +46,83 @@
...
-    @Before
-    public void setUp() {
-		this.events = new EventBus<>();
-		this.events.on(ProgressBarEvent.class).bind(evt -> System.out.println(evt.message));
-	}
-
 	@Test
 	public void testProgressBar() {
+		EventBus<Event> events = new EventBus<>();
+		List<Object[]> testContents = Arrays.asList(
```
Comment (RX14) [inline src/test/java/nova/core/util/ProgressBarTest.java]: This seems uglier than just calling the methods directly.

File: src/test/java/nova/core/util/ProgressBarTest.java [hunk 6]
```diff
@@ -40,30 +46,83 @@
...
-		progressBar.step(ProgressBarTest.class);
-		progressBar.step("Progress bar testing mod message", TestMod.class);
-		progressBar.step(TestMod.class, "Progress bar testing mod message");
-		progressBar.step(TestMod.class);
+		for (Object[] testArg : testContents) {
+			step(progressBar, testArg);
+		}
 		progressBar.finish();
+
+		List<String> expectedResults = testContents.stream().map(ProgressBarTest::convert).collect(Collectors.toList());
```
Comment (RX14) [inline src/test/java/nova/core/util/ProgressBarTest.java]: This should be manually-entered, there could be bugs in your `convert` impl.

File: src/test/java/nova/core/util/ProgressBarTest.java [hunk 7]
```diff
@@ -40,30 +46,83 @@
...
+
+		assertThat(testResults.size()).isEqualTo(testContents.size());
+		assertThat(testResults.size()).isEqualTo(expectedResults.size());
+
+		for (int i = 0; i < testContents.size(); i++)
+			assertThat(testResults.get(i)).isEqualTo(expectedResults.get(i));
```
Comment (RX14) [inline src/test/java/nova/core/util/ProgressBarTest.java]: This can be done better using assertj collection assertions. For example 

```java
assertThat(foo).containsExactly(
    "Progress bar testing generic String message",
    "Progress bar testing class message: ProgressBarTest"
);
```

File: src/test/java/nova/core/util/ProgressBarTest.java [hunk 8]
```diff
@@ -40,30 +46,83 @@
...
+			new Object[]{(Class) null},
+			new Object[]{ProgressBarTest.class, null},
+			new Object[]{null, ProgressBarTest.class});
+		List<String> testResults = new LinkedList<>();
+		events.on(TestProgressBarEvent.class).bind(evt -> testResults.add(evt.message));
 		ProgressBar progressBar = new AbstractProgressBar() {
 			@Override
 			protected void stepImpl(String message) {
-				events.publish(new ProgressBarEvent(message));
+				events.publish(new TestProgressBarEvent(message));
```
Comment (RX14) [inline src/test/java/nova/core/util/ProgressBarTest.java]: Why not skip the event bus and add it to the list directly?
